### DISCUSSION TITLE
The custom rule for windows__eventchannet is not assigned

### DISCUSSION THREAD

**Discussion by Kobrik1** [wazuh/wazuh#25405] [Q&A]
Hi, I am unable to assign a custom rule for the log from windows_eventchannel.

This is a log:

{"win":{"system":{"providerName":"Service Control Manager","providerGuid":"{555908d1-a6d7-4695-8e1e-26931d2012f4}","eventSourceName":"Service Control Manager","eventID":"7036","version":"0","level":"4","task":"0","opcode":"0","keywords":"0x8080000000000000","systemTime":"2024-08-26T09:12:55.7670535Z","eventRecordID":"99273","processID":"768","threadID":"5068","channel":"System","computer":"dcx0nt.com.local","severityValue":"INFORMATION","message":"\"The Zabbix Agent 2 service entered the stopped state.\""},"eventdata":{"param1":"Zabbix Agent 2","param2":"stopped","binary":"5A006100620062006900780020004100670065006E007400200032002F0031000000"}}}

Here is the rule created:

<group name="windows,service_control,">

  <rule id="130001" level="0">
    <field name="win.system.providerName">Service Control Manager</field>
    <options>no_full_log</options>
    <description>Service Control Manager, $(win.system.computer), stopping service $(win.eventdata.param1)</description>
  </rule>

  <rule id="130002" level="9">
    <if_sid>130001</if_sid>
    <field name="win.system.eventID">7036</field>
    <field name="win.eventdata.param1">Zabbix Agent|ESET Service|ESET Management Agent|ESET Forwarder</field>
    <field name="win.eventdata.param2">stopped</field>
    <field name="win.system.computer"> dcx0nt.com.local </field>
    <description>$(win.system.computer), zastavena služba $(win.eventdata.param1)</description>
    <group>zabbix,</group>
  </rule>
 
</group>

Very disturbing is the fact that the log passes the Ruleset Test without any problems:

**Phase 1: Completed pre-decoding.
	full event: '{"win":{"system":{"providerName":"Service Control Manager","providerGuid":"{555908d1-a6d7-4695-8e1e-26931d2012f4}","eventSourceName":"Service Control Manager","eventID":"7036","version":"0","level":"4","task":"0","opcode":"0","keywords":"0x8080000000000000","systemTime":"2024-08-26T09:12:55.7670535Z","eventRecordID":"99273","processID":"768","threadID":"5068","channel":"System","computer":" dcx0nt.com.local","severityValue":"INFORMATION","message":"\"The Zabbix Agent 2 service entered the stopped state.\""},"eventdata":{"param1":"Zabbix Agent 2","param2":"stopped","binary":"5A006100620062006900780020004100670065006E007400200032002F0031000000"}}}'

**Phase 2: Completed decoding.
	name: 'json'
	win.eventdata.binary: '5A006100620062006900780020004100670065006E007400200032002F0031000000'
	win.eventdata.param1: 'Zabbix Agent 2'
	win.eventdata.param2: 'stopped'
	win.system.channel: 'System'
	win.system.computer: 'dcx0nt.com.local'
	win.system.eventID: '7036'
	win.system.eventRecordID: '99273'
	win.system.eventSourceName: 'Service Control Manager'
	win.system.keywords: '0x8080000000000000'
	win.system.level: '4'
	win.system.message: '"The Zabbix Agent 2 service entered the stopped state."'
	win.system.opcode: '0'
	win.system.processID: '768'
	win.system.providerGuid: '{555908d1-a6d7-4695-8e1e-26931d2012f4}'
	win.system.providerName: 'Service Control Manager'
	win.system.severityValue: 'INFORMATION'
	win.system.systemTime: '2024-08-26T09:12:55.7670535Z'
	win.system.task: '0'
	win.system.threadID: '5068'
	win.system.version: '0'

**Phase 3: Completed filtering (rules).
	id: '130002'
	level: '9'
	description: 'dcx0nt.com.local, zastavena služba Zabbix Agent 2'
	groups: '["windows","service_control","zabbix"]'
	firedtimes: '1'
	mail: 'false'
**Alert to be generated.

So far, everything looks great, but it's not happening. As you can see below from the JSON log from wazuh, the rule does not connect despite a positive test:

{
  "_index": "wazuh-archives-4.x-2024.08.26",
  "_id": "otH1jZEBkjvmiqrowRFH",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "10.1.1.10",
      "name": "dcx0nt",
      "id": "021"
    },
    "manager": {
      "name": "wazuh-server"
    },
    "data": {
      "win": {
        "eventdata": {
          "binary": "5A006100620062006900780020004100670065006E007400200032002F0031000000",
          "param1": "Zabbix Agent 2",
          "param2": "stopped"
        },
        "system": {
          "eventID": "7036",
          "eventSourceName": "Service Control Manager",
          "keywords": "0x8080000000000000",
          "providerGuid": "{555908d1-a6d7-4695-8e1e-26931d2012f4}",
          "level": "4",
          "channel": "System",
          "opcode": "0",
          "message": "\"The Zabbix Agent 2 service entered the stopped state.\"",
          "version": "0",
          "systemTime": "2024-08-26T09:12:55.7670535Z",
          "eventRecordID": "99273",
          "threadID": "5068",
          "computer": "dcx0nt.com.local",
          "task": "0",
          "processID": "768",
          "severityValue": "INFORMATION",
          "providerName": "Service Control Manager"
        }
      }
    },
    "location": "EventChannel",
    "decoder": {
      "name": "windows_eventchannel"
    },
    "id": "1724663576.437586609",
    "full_log": "{\"win\":{\"system\":{\"providerName\":\"Service Control Manager\",\"providerGuid\":\"{555908d1-a6d7-4695-8e1e-26931d2012f4}\",\"eventSourceName\":\"Service Control Manager\",\"eventID\":\"7036\",\"version\":\"0\",\"level\":\"4\",\"task\":\"0\",\"opcode\":\"0\",\"keywords\":\"0x8080000000000000\",\"systemTime\":\"2024-08-26T09:12:55.7670535Z\",\"eventRecordID\":\"99273\",\"processID\":\"768\",\"threadID\":\"5068\",\"channel\":\"System\",\"computer\":\"dcx0nt.com.local\",\"severityValue\":\"INFORMATION\",\"message\":\"\\\"The Zabbix Agent 2 service entered the stopped state.\\\"\"},\"eventdata\":{\"param1\":\"Zabbix Agent 2\",\"param2\":\"stopped\",\"binary\":\"5A006100620062006900780020004100670065006E007400200032002F0031000000\"}}}",
    "timestamp": "2024-08-26T09:12:56.779+0000"
  },
  "fields": {
    "timestamp": [
      "2024-08-26T09:12:56.779Z"
    ]
  },
  "highlight": {
    "data.win.system.eventID": [
      "@opensearch-dashboards-highlighted-field@7036@/opensearch-dashboards-highlighted-field@"
    ],
    "full_log": [
      "{\"win\":{\"system\":{\"providerName\":\"Service Control Manager\",\"providerGuid\":\"{555908d1-a6d7-4695-8e1e-26931d2012f4}\",\"eventSourceName\":\"Service Control Manager\",\"eventID\":\"7036\",\"version\":\"0\",\"level\":\"4\",\"task\":\"0\",\"opcode\":\"0\",\"keywords\":\"0x8080000000000000\",\"systemTime\":\"2024-08-26T09:12:55.7670535Z\",\"eventRecordID\":\"99273\",\"processID\":\"768\",\"threadID\":\"5068\",\"channel\":\"System\",\"computer\":\"dcx0nt.com.local\",\"severityValue\":\"INFORMATION\",\"message\":\"\\\"The @opensearch-dashboards-highlighted-field@Zabbix@/opensearch-dashboards-highlighted-field@ Agent 2 service entered the stopped state.\\\"\"},\"eventdata\":{\"param1\":\"@opensearch-dashboards-highlighted-field@Zabbix@/opensearch-dashboards-highlighted-field@ Agent 2\",\"param2\":\"stopped\",\"binary\":\"5A006100620062006900780020004100670065006E007400200032002F0031000000\"}}}"
    ]
  },
  "sort": [
    1724663576779
  ]
} {
  "_index": "wazuh-archives-4.x-2024.08.26",
  "_id": "otH1jZEBkjvmiqrowRFH",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "1.1.1.1",
      "name": "dcx0nt.com.local",
      "id": "021"
    },
    "manager": {
      "name": "wazuh-server"
    },
    "data": {
      "win": {
        "eventdata": {
          "binary": "5A006100620062006900780020004100670065006E007400200032002F0031000000",
          "param1": "Zabbix Agent 2",
          "param2": "stopped"
        },
        "system": {
          "eventID": "7036",
          "eventSourceName": "Service Control Manager",
          "keywords": "0x8080000000000000",
          "providerGuid": "{555908d1-a6d7-4695-8e1e-26931d2012f4}",
          "level": "4",
          "channel": "System",
          "opcode": "0",
          "message": "\"The Zabbix Agent 2 service entered the stopped state.\"",
          "version": "0",
          "systemTime": "2024-08-26T09:12:55.7670535Z",
          "eventRecordID": "99273",
          "threadID": "5068",
          "computer": "dcx0nt.com.local",
          "task": "0",
          "processID": "768",
          "severityValue": "INFORMATION",
          "providerName": "Service Control Manager"
        }
      }
    },
    "location": "EventChannel",
    "decoder": {
      "name": "windows_eventchannel"
    },
    "id": "1724663576.437586609",
    "full_log": "{\"win\":{\"system\":{\"providerName\":\"Service Control Manager\",\"providerGuid\":\"{555908d1-a6d7-4695-8e1e-26931d2012f4}\",\"eventSourceName\":\"Service Control Manager\",\"eventID\":\"7036\",\"version\":\"0\",\"level\":\"4\",\"task\":\"0\",\"opcode\":\"0\",\"keywords\":\"0x8080000000000000\",\"systemTime\":\"2024-08-26T09:12:55.7670535Z\",\"eventRecordID\":\"99273\",\"processID\":\"768\",\"threadID\":\"5068\",\"channel\":\"System\",\"computer\":\"dcx0nt.com.local\",\"severityValue\":\"INFORMATION\",\"message\":\"\\\"The Zabbix Agent 2 service entered the stopped state.\\\"\"},\"eventdata\":{\"param1\":\"Zabbix Agent 2\",\"param2\":\"stopped\",\"binary\":\"5A006100620062006900780020004100670065006E007400200032002F0031000000\"}}}",
    "timestamp": "2024-08-26T09:12:56.779+0000"
  },
  "fields": {
    "timestamp": [
      "2024-08-26T09:12:56.779Z"
    ]
  },
  "highlight": {
    "data.win.system.eventID": [
      "@opensearch-dashboards-highlighted-field@7036@/opensearch-dashboards-highlighted-field@"
    ],
    "full_log": [
      "{\"win\":{\"system\":{\"providerName\":\"Service Control Manager\",\"providerGuid\":\"{555908d1-a6d7-4695-8e1e-26931d2012f4}\",\"eventSourceName\":\"Service Control Manager\",\"eventID\":\"7036\",\"version\":\"0\",\"level\":\"4\",\"task\":\"0\",\"opcode\":\"0\",\"keywords\":\"0x8080000000000000\",\"systemTime\":\"2024-08-26T09:12:55.7670535Z\",\"eventRecordID\":\"99273\",\"processID\":\"768\",\"threadID\":\"5068\",\"channel\":\"System\",\"computer\":\"dcx0nt.com.local\",\"severityValue\":\"INFORMATION\",\"message\":\"\\\"The @opensearch-dashboards-highlighted-field@Zabbix@/opensearch-dashboards-highlighted-field@ Agent 2 service entered the stopped state.\\\"\"},\"eventdata\":{\"param1\":\"@opensearch-dashboards-highlighted-field@Zabbix@/opensearch-dashboards-highlighted-field@ Agent 2\",\"param2\":\"stopped\",\"binary\":\"5A006100620062006900780020004100670065006E007400200032002F0031000000\"}}}"
    ]
  },
  "sort": [
    1724663576779
  ]
}

I have the impression that the assignment to 0575-win-base_rules.xml does not work well in this case. Because the base rule

  <rule id="60000" level="0">
    <category>ossec</category>
    <decoded_as>windows_eventchannel</decoded_as>
    <field name="win.system.providerName"></field>
    <options>no_full_log</options>
    <description>Group of windows rules.</description>
  </rule>

should match the log I provided.

I really ask for advice. Is this my mistake? I personally see it more as a bug in wazuh.

**Comment by Stuti3097** [comment]
Hi @Kobrik1 

To test Windows EventChannel logs using `wazuh-logtest` or `ruletest`, you need to temporarily modify rule 60000 in the file `/var/ossec/ruleset/rules/0575-win-base_rules.xml`. By default, these logs are decoded as `windows_eventchannel`, which can prevent proper testing. To make them testable, comment out the `<category>` and `<decoded_as>` lines, and replace with `<decoded_as>json</decoded_as>`:

```xml
<rule id="60000" level="2">
    <!-- <category>ossec</category> -->
    <!-- <decoded_as>windows_eventchannel</decoded_as> -->
    <decoded_as>json</decoded_as>
    <field name="win.system.providerName">\.+</field>
    <options>no_full_log</options>
    <description>Group of Windows rules.</description>
</rule>
```

This change allows the logs to be processed by the test tools. After applying the modification, test the logs using `wazuh-logtest` or `ruletest`. If the log matches rule 61100, you can write a custom rule based on that match. For example:

```xml
<rule id="100002" level="3">
    <if_sid>61100</if_sid>
    <field name="win.eventdata.param1">\.*Zabbix\.*</field>
    <description>$(win.system.message)</description>
    <group>windows,service_control,zabbix</group>
</rule>
```

**Once testing is complete, revert rule 60000 to its original form and restart the Wazuh manager using:**

```bash
sudo systemctl restart wazuh-manager
```


<img width="1574" height="806" alt="image" src="https://github.com/user-attachments/assets/45a067a7-202a-4a4d-a20f-d9da2e31e2bc" />
<img width="1561" height="785" alt="image" src="https://github.com/user-attachments/assets/a83e468c-e6e6-46de-b654-91547289c7ca" />
